lib/execute/src/memory.rs

```diff
@@ -53,15 +53,29 @@ impl LinearMemory {
             Some(new_pages) => new_pages,
             None => return None,
         };
+        if let Some(val) = self.maximum {
+            if new_pages > val {
+                return None;
+            }
+        }
 
         let prev_pages = self.current;
+        let new_bytes = (new_pages * PAGE_SIZE) as usize;
+
+        if self.mmap.len() < new_bytes {
+            assert!(self.maximum.is_none());
+            let mut new_mmap = memmap::MmapMut::map_anon(new_bytes).unwrap();
+            new_mmap.copy_from_slice(&self.mmap);
+            self.mmap = new_mmap;
```

Comment:
Hm, I thought that heap base in generated code could be cached and because of this, my idea was to implement `LinearMemory` with a stable base address. That is, allocate virtual memory up to the maximum (if any, otherwise 4GB) and then "unprotect" it upon growing. 

With the proposed change the base becomes unstable and cannot longer be cached.

Reply:
Cranelift currently supports two styles of heaps, static and dynamic, which are meant to correspond to wasm linear memories with and without a maximum, respectively. This needs more work in wasmtime though.

Reply:
yeah, that's why I started with that approach. I guess the static heap also should be more efficient, right?

Reply:
Yes, that's right. Also, the dynamic heap won't support threads. So you're right that we should mainly focus on static.

Reply:
I added a comment clarifying the role of this code.